File: packages/shared-types/src/socket-events.ts
/**
 * Socket event names — must match app/Constants/SocketEvents.php (Laravel).
 * Frontend/Dart copies these string constants manually until codegen ships.
 */
export const SOCKET_EVENTS = {
  DRIVER_LOCATION_UPDATE: 'driver:location:update',
  DRIVER_STATUS_UPDATE: 'driver:status:update',
  RIDE_ACCEPT: 'ride:accept',
  RIDE_DECLINE: 'ride:decline',
  RIDE_ARRIVED: 'ride:arrived',
  RIDE_CASH_CONFIRM: 'ride:cash:confirm',
  RIDE_COMPLETE: 'ride:complete',
  RIDE_CANCEL: 'ride:cancel',
  SERVER_RIDE_DISPATCHED: 'server:ride:dispatched',
  SERVER_RIDE_CANCELLED: 'server:ride:cancelled',
  SERVER_RIDE_TIMEOUT: 'server:ride:timeout',
  SERVER_RIDE_ACCEPTED: 'server:ride:accepted',
  SERVER_DRIVER_LOCATION: 'server:driver:location',
  SERVER_DRIVER_ARRIVED: 'server:driver:arrived',
  SERVER_RIDE_STARTED: 'server:ride:started',
  SERVER_RIDE_COMPLETED: 'server:ride:completed',
  SERVER_DRIVER_CANCELLED: 'server:driver:cancelled',
  ADMIN_SOS_ALERT: 'admin:sos:alert',
  ADMIN_DASHBOARD: 'admin:dashboard',
} as const;

export type SocketEventName = typeof SOCKET_EVENTS[keyof typeof SOCKET_EVENTS];

File: apps/passenger-app/lib/constants/socket_events.dart
/// Mirror of apps/api/app/Constants/SocketEvents.php. Keep in sync when
/// the server gains a new event; the build will break if either side
/// drifts, which is intentional.
class SocketEvents {
  SocketEvents._();

  // Driver → server (driver app only; listed here for parity)
  static const String driverLocationUpdate = 'driver:location:update';
  static const String driverStatusUpdate = 'driver:status:update';
  static const String rideAccept = 'ride:accept';
  static const String rideDecline = 'ride:decline';
  static const String rideArrived = 'ride:arrived';
  static const String rideCashConfirm = 'ride:cash:confirm';
  static const String rideComplete = 'ride:complete';

  // Passenger → server (passenger app only; listed here for parity)
  static const String rideCancel = 'ride:cancel';

  // Server → driver (driver app only)
  static const String serverRideDispatched = 'server:ride:dispatched';
  static const String serverRideCancelled = 'server:ride:cancelled';
  static const String serverRideTimeout = 'server:ride:timeout';

  // Server → passenger (passenger app)
  static const String serverRideAccepted = 'server:ride:accepted';
  static const String serverDriverLocation = 'server:driver:location';
  static const String serverDriverArrived = 'server:driver:arrived';
  static const String serverRideStarted = 'server:ride:started';
  static const String serverRideCompleted = 'server:ride:completed';
  static const String serverDriverCancelled = 'server:driver:cancelled';

  // Admin
  static const String adminSosAlert = 'admin:sos:alert';
  static const String adminDashboard = 'admin:dashboard';
}
